File: api-dot-net/Controllers/AdminController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using CjsApi.Repositories.ProblemRepository;
using CjsApi.Repositories.SubmissionRepository;
using CjsApi.Repositories.UserRepository;
using CjsApi.Models;

namespace CjsApi.Controllers
{
    [Route("api/admin/stats")]
    [ApiController]
    public class AdminController : ControllerBase
    {
        private readonly IProblemRepository _problemRepository;
        private readonly ISubmissionRepository _submissionRepository;
        private readonly IUserRepository _userRepository;

        public AdminController(
            IProblemRepository problemRepository, 
            ISubmissionRepository submissionRepository,
            IUserRepository userRepository)
        {
            _problemRepository = problemRepository;
            _submissionRepository = submissionRepository;
            _userRepository = userRepository;
        }

        [HttpGet("overview")]
        public async Task<IActionResult> GetStatsOverview()
        {
            var problemCount = await _problemRepository.CountAsync();
            var submissionCount = await _submissionRepository.CountAsync();
            var userCount = await _userRepository.CountAsync();

            return Ok(new
            {
                success = true,
                message = "Stats retrieved",
                data = new
                {
                    totalProblems = problemCount,
                    totalSubmissions = submissionCount,
                    totalUsers = userCount
                }
            });
        }
    }
}

File: java-api/cjs-service/src/main/java/com/cjs/cjs_service/controller/AdminController.java
package com.cjs.cjs_service.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.cjs.cjs_service.dto.response.ApiResponseDto;
import com.cjs.cjs_service.repository.ProblemRepository;
import com.cjs.cjs_service.repository.SubmissionRepository;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/admin/stats")
public class AdminController {

    private final ProblemRepository problemRepository;
    private final SubmissionRepository submissionRepository;

    public AdminController(ProblemRepository problemRepository, SubmissionRepository submissionRepository) {
        this.problemRepository = problemRepository;
        this.submissionRepository = submissionRepository;
    }

    @GetMapping("/overview")
    public ResponseEntity<ApiResponseDto<Map<String, Long>>> getStatsOverview() {
        long problemCount = problemRepository.count();
        long submissionCount = submissionRepository.count();

        Map<String, Long> stats = new HashMap<>();
        stats.put("totalProblems", problemCount);
        stats.put("totalSubmissions", submissionCount);

        return ResponseEntity.ok(
            new ApiResponseDto<>("Stats overview retrieved", stats, true)
        );
    }
}
